lib/analytics.js

```diff
@@ -114,13 +126,18 @@ Analytics.prototype.init = Analytics.prototype.initialize = function(settings, o
     if (!Integration) delete settings[name];
   }, settings);
 
-  // add integrations
-  each(function(opts, name) {
-    var Integration = self.Integrations[name];
-    var integration = new Integration(clone(opts));
-    self.log('initialize %o - %o', name, opts);
-    self.add(integration);
-  }, settings);
+  if (doNotTrack) {
+    // Clear manually added integrations
+    this._integrations = {};
```

Comment:
Note that this will remove the `Segment.io` integration, so the data will not make it to Segment. This renders `context.doNotTrack` useless because nothing will receive it